frontend/lib/norent/components/subscribe.tsx

```diff
@@ -0,0 +1,136 @@
+import React from "react";
+import { Trans, t } from "@lingui/macro";
+import { li18n } from "../../i18n-lingui";
+import { StaticImage } from "../../ui/static-image";
+import { getImageSrc } from "../homepage";
+
+type SubscribeProps = {};
+
+type SubscribeState = {
+  email: string;
+  success: boolean;
+  response: string;
+};
+
+class Subscribe extends React.Component<SubscribeProps, SubscribeState> {
+  constructor(props: SubscribeProps) {
+    super(props);
+    this.state = {
+      email: "",
+      success: false,
+      response: "",
+    };
+  }
+
+  handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ email: e.target.value });
+  };
+
+  handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const email = this.state.email || null;
+    const locale = li18n.language;
+
+    if (!email) {
+      this.setState({
+        response: li18n._(t`Please enter an email address!`),
+      });
+      return;
+    }
+
+    fetch("/mailchimp/subscribe", {
+      method: "POST",
+      body: `email=${encodeURIComponent(
+        email
+      )}&language=${locale}&source=norent`,
+      headers: {
+        "Content-Type": "application/x-www-form-urlencoded",
+      },
+    })
+      .then((result) => result.json())
+      .then((result) => {
+        if (result.status === 200) {
+          this.setState({
+            success: true,
+            response: li18n._(t`All set! Thanks for subscribing!`),
+          });
+        } else if (result.errorCode === "INVALID_EMAIL") {
+          this.setState({
+            response: li18n._(t`Oops! That email is invalid.`),
+          });
+        } else {
+          window &&
+            window.Rollbar &&
+            window.Rollbar.error(
+              `Mailchimp email signup responded with error code ${result.errorCode}.`
+            );
+          this.setState({
+            response: li18n._(
+              t`Oops! A network error occurred. Try again later.`
+            ),
+          });
+        }
+      })
+      .catch((err) => {
+        this.setState({
+          response: li18n._(
+            t`Oops! A network error occurred. Try again later.`
+          ),
+        });
+      });
+  };
+
+  render() {
+    return (
+      <div>
```

Comment:
Since this form currently requires JavaScript, it might be worth wrapping it in a `<SimpleProgressiveEnhancement>` so that it doesn't render on clients that haven't enabled JS/are in compatibility mode.  Ideally it'd be nice if this form didn't require JS, too, so maybe we should file an issue that documents it?  Unless you feel strongly that it's not core functionality to the site, which I could also understand.

Reply:
great! I'm gonna wrap all of the email sign-up related content in this component so none of it shows up in safe mode.

Reply:
And I think we should file a bug about it!